tutorials/pubsub-spring-dedup-messages/index.md

```diff
@@ -0,0 +1,263 @@
+```java
+  // The Supplier Bean makes the function a valid Spring Cloud Stream source. It
+  // sends messages to a Cloud Pub/Sub topic configured with the binding name
+  // `sendMessagesForDeduplication-out-0` in application.properties.
+  @Bean
+  Supplier<Flux<Message<String>>> sendMessagesForDeduplication(
+    final EmitterProcessor<Message<String>> frontEndListener) {
+    return () -> frontEndListener;
+  }
+```
+
+### Specify a Cloud Pub/Sub topic for the source
+Spring can find the Cloud Pub/Sub topic that your code should publish data to when you provide a topic name for your source in `application.properties` using an output binder. Here, the output binder's binding name is `sendMessagesForDeduplication-out-0`. For more information, see [Binding and Binding Names].
+
+In [application.properties](pubsub-spring/src/main/resources/application.properties):
+
+[embedmd]:# (pubsub-spring/src/main/resources/application.properties /.*Data going/ /=topicToDataflow/)
+```properties
+# Data going to Cloud Pub/Sub from a Spring Cloud Stream source defined in the
+# Supplier bean `sendMessagesForDeduplication`. The application will
+# create the Cloud Pub/Sub topic `topicToDataflow` if it does not exist.
+spring.cloud.stream.bindings.sendMessagesForDeduplication-out-0.destination=topicToDataflow
+```
+
+Similarly, your application will know how to receive data from a Cloud Pub/Sub subscription if you specify 1). a Spring Cloud Stream sink and 2). a pair of Cloud Pub/Sub topic and subscription.
+
+### Specify a Spring Cloud Stream sink
+Spring can recognize a Spring Cloud Stream sink as a Consumer bean.
+
+In [App.java](pubsub-spring/src/main/java/com/google/example/App.java):
+
+[embedmd]:# (pubsub-spring/src/main/java/com/google/example/App.java java /  \/\/ The Consumer Bean/ /}/)
+```java
+  // The Consumer Bean makes the function a valid Spring Cloud Stream sink. It
+  // receives messages from a Cloud Pub/Sub subscription configured with the binding
+  // name `receiveDedupedMessagesFromDataflow-in-0` in application.properties.
+  @Bean
+  Consumer<Message<String>> receiveDedupedMessagesFromDataflow() {
+    return msg -> {
+      System.out.println("\tDE-DUPED message: \"" + msg.getPayload() + "\".");
+    }
+```
+
+### Specify a Cloud Pub/Sub topic and subscription for the sink
+Spring can find the Cloud Pub/Sub subscription that your code should receive data from when you provide a subscription name for your sink in `application.properties` using an input binder. Here, the input binder's binding name is `receiveDedupedMessagesFromDataflow-in-0`. Because a subscription cannot exist without a topic, a topic must also be specified. The input binder's destination is your Pub/Sub topic and its consumer group is your Pub/Sub subscription. Note that only input bindings have consumer groups. For more information, see [Common Binding Properties].
```

Comment:
Similar comment as before. It may be useful to start this sentence with the section's objective.

Is the `receiveDedupedMessagesFromDataflow-in-0` binder relevant only if you're following the tutorial or is it a keyword that applies to all applications can use to get the subscription name? This also applies to other things in code font, like `receiveDedupedMessagesFromDataflow` and `topicFromDataflow`.